### ISSUE TITLE
System.TypeLoadException: Method 'Build' in type 'Microsoft.Framework.Configuration.ConfigurationBuilder'

### ISSUE THREAD

**Issue by npateman** [aspnet/Home#884]
Updating to Beta 7 and assuring that all of my defendencies are also beta 7, I get the following error,

---

System.TypeLoadException: Method 'Build' in type 'Microsoft.Framework.Configuration.ConfigurationBuilder' from assembly
'Microsoft.Framework.Configuration, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementati
on.
   at Microsoft.AspNet.Hosting.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider service
Provider)
   at Microsoft.Dnx.ApplicationHost.Program.ExecuteMain(DefaultHost host, String applicationName, String[] args)
   at Microsoft.Dnx.ApplicationHost.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider service
Provider)
   at Microsoft.Dnx.Host.Bootstrapper.RunAsync(List`1 args, IRuntimeEnvironment env, FrameworkName targetFramework)
   at Microsoft.Dnx.Host.RuntimeBootstrapper.ExecuteAsync(String[] args, FrameworkName targetFramework)
   at Microsoft.Dnx.Host.RuntimeBootstrapper.Execute(String[] args, FrameworkName targetFramework)

---

I've checked the breaking changes list but can't see anything relating to this, but have read an old post suggesting that it is due to framework mismatches.

My Config file has the following,

```
"dependencies": {
    "Microsoft.Framework.Configuration": "1.0.0-beta7",
    "Microsoft.Framework.Configuration.Json": "1.0.0-beta7",
    "Microsoft.AspNet.Mvc": "6.0.0-beta7",
    "Microsoft.AspNet.Mvc.TagHelpers": "6.0.0-beta7",
    "Microsoft.AspNet.StaticFiles": "1.0.0-beta7",
    "Microsoft.AspNet.Tooling.Razor": "1.0.0-beta7",
    "Microsoft.Framework.Logging": "1.0.0-beta7",
    "Microsoft.Framework.Logging.Console": "1.0.0-beta7",
    "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.0.0-beta7",
    "Microsoft.AspNet.Mvc.WebApiCompatShim": "6.0.0-beta7",
    "System.Security.Cryptography.DeriveBytes": "4.0.0-beta-23224",
    "MyLibrary": "1.0.0-*",
    "Kestrel": "1.0.0-beta7",
    "dnx-clr-win-x64": "1.0.0-beta7"
},

"frameworks": {
    "dnxcore5": {
        "dependencies": {
            "System.Runtime.Serialization.Json": "4.0.0-beta-23110",
            "System.Runtime.Extensions": "4.0.11-beta-23225",
            "System.Collections": "4.0.11-beta-23225",
            "System.Runtime": "4.0.21-beta-23225",
            "Microsoft.CSharp": "4.0.1-beta-23225",
            "System.Private.Networking": "4.0.1-beta-23225",
            "System.Net.Security": "4.0.0-beta-23301"
        }
    }
},
```

The one for MyLibrary is

```
"dependencies": {
  "dnx-clr-win-x64": "1.0.0-beta7"
},

"frameworks": {
"dnxcore5": {
  "dependencies": {
    "System.Collections": "4.0.11-beta-23225",
    "System.Runtime": "4.0.21-beta-23225",
    "Microsoft.CSharp": "4.0.1-beta-23225"
  }
}
}
```

**Comment by npateman** [comment]
<img width="308" alt="buildmethod" src="https://cloud.githubusercontent.com/assets/1122645/9659196/efddffda-5246-11e5-9b25-0a579f9e4330.PNG">

That's Reflector showing the actual library that is in the approot folder, which I assume it should be using but can't find the Build method that is clearly there (highlighted)

**Comment by davidfowl** [comment]
I'm not sure what issue you're running into but this reference in your project.json is bogus `    "dnx-clr-win-x64": "1.0.0-beta7"`

**Comment by npateman** [comment]
I was told that I needed that if I want to compile for x64 dnxcore.  I was using the beta 5 one just fine so assumed that as it found a beta 7 reference then that was okay.  I can see now that the dependency folder is empty so it's not actually downloaded the library.

Either way, removing that reference and compiling for x86 still results in the same error.

**Comment by davidfowl** [comment]
Can you provide a runnable sample on github? The only possible thing would be a dependency mismatch but there's too little information to tell what's going on. Look at your `project.lock.json` to see what actual dependency versions got resolved.

**Comment by npateman** [comment]
Can I invite you to a private repo perhaps?  It's not public yet.

**Comment by davidfowl** [comment]
Did you look at the `project.lock.json` file?

**Comment by npateman** [comment]
It's over 6000 lines long lol, not entirely sure what I should be looking at in there. Microsoft.Framework.Configuration is listed in there.

**Comment by davidfowl** [comment]
Is anything beta6? Is everything beta7? Do a search in the file for the versions...

**Comment by npateman** [comment]
okay, so i'm doing a text search for beta(n)

beta1 comes up with a few, for example

Microsoft.CodeAnalysis.Analyzers/1.1.0-beta1-20150812-01
Microsoft.CodeAnalysis.Common/1.1.0-beta1-20150812-01
Microsoft.CodeAnalysis.CSharp/1.1.0-beta1-20150812-01

...

Everything else is beta7, I think the above were added at compile time as I don't reference them myself.

**Comment by davidfowl** [comment]
I'm pretty sure that's not true. The fact is that:

```
"Kestrel": "1.0.0-beta7",
```

Does not exist and has been renamed to Microsoft.AspNet.Server.Kestrel. https://github.com/aspnet/Announcements/issues/53

**Comment by npateman** [comment]
oh let me try changing that to see what happens, nice it doesn't tell me that.

What do you mean by not true though?  I just searched the document, and nothing came up, it's resolving that dependency and not giving any errors, must be a bug in Visual Studio, but it was doing the same for the 64 bit libs as discovered earlier.

**Comment by npateman** [comment]
After changing that reference the issue persists unfortunately.

**Comment by npateman** [comment]
Okay so with that oddity in mind, I'm going through each package folder after publishing to see which ones are missing an actual library in the folder,

Microsoft.Bcl.Build
Microsoft.CodeAnalysis.Analyzers

The above 2 have no lib, and the following ones may be odd, not quite sure

System.Reflection.Metadata\1.1.0-alpha-00009
icrosoft.Net.Http\2.2.22

**Comment by davidfowl** [comment]
There are several things in this thread that are conflicting so I'm having a hard time figuring out exactly what you're doing. 

> I was told that I needed that if I want to compile for x64 dnxcore.

They probably meant run not compile with (I hope).

> That's Reflector showing the actual library that is in the approot folder, which I assume it should be using but can't find the Build method that is clearly there (highlighted)

What approot folder? Did you publish?

> I just searched the document, and nothing came up, it's resolving that dependency and not giving any errors

What version of Kestrel got resolved?

**Comment by npateman** [comment]
Well that's the ultimate goal, but the issue at current is after updating to beta 7, I got this error.

That approot that's created by visual studio, it puts an the packages with it, just published to a folder and looked inside.

-----Original Message-----

**Comment by npateman** [comment]
Sorry I missed a few of your questions as I replied from my mobile... Let me try again,

There are several things in this thread that are conflicting so I'm having a hard time figuring out exactly what you're doing.

    Compiling for DNXCore x64, as I want to run it on Linux, which displays red error text when attempting to install the coreclr as x86 stating that it must be x64.  I want to run this core project on anything I can, ideally!

They probably meant run not compile with (I hope).

    Well yeah, that's just the way I have phrased it.  It needs to be referenced with the project.

What approot folder? Did you publish?

    Just the one that Visual Studio created during publishing to a local folder.  That folder works just fine on Windows Server with ASP 5 installed manually, well it did prior to Beta 7.

What version of Kestrel got resolved?

    Well now, beta 7, previously it said it had resolved but the published library wasn't there.  No compiler errors etc. which leads me to believe it's a bug with Visual Studio as I had no idea anything was wrong there.

**Comment by LeoNatanian** [comment]
If I may provide some supplemental information, I too am facing the same issue when updating my project.json file to use AspNet.MVC 6.0-beta 7.  

To demonstrate the issue I did the following

1. I created a new default project template.
2. I opened project.json file, and changed the value of  "Microsoft.AspNet.Mvc": "6.0.0-beta5", to "Microsoft.AspNet.Mvc": "6.0.0-beta7",

Re-build, run and I now received the issue.  This is my entire project.json dependency section for reference. Note: I purposely did not change other items to beta7 from beta5 so I can replicate this error above.

  "dependencies": {
    "EntityFramework.SqlServer": "7.0.0-beta5",
    "EntityFramework.Commands": "7.0.0-beta5",
    "Microsoft.AspNet.Mvc": "6.0.0-beta7",
    "Microsoft.AspNet.Mvc.TagHelpers": "6.0.0-beta5",
    "Microsoft.AspNet.Authentication.Cookies": "1.0.0-beta5",
    "Microsoft.AspNet.Authentication.Facebook": "1.0.0-beta5",
    "Microsoft.AspNet.Authentication.Google": "1.0.0-beta5",
    "Microsoft.AspNet.Authentication.MicrosoftAccount": "1.0.0-beta5",
    "Microsoft.AspNet.Authentication.Twitter": "1.0.0-beta5",
    "Microsoft.AspNet.Diagnostics": "1.0.0-beta5",
    "Microsoft.AspNet.Diagnostics.Entity": "7.0.0-beta5",
    "Microsoft.AspNet.Identity.EntityFramework": "3.0.0-beta5",
    "Microsoft.AspNet.Server.IIS": "1.0.0-beta5",
    "Microsoft.AspNet.Server.WebListener": "1.0.0-beta5",
    "Microsoft.AspNet.StaticFiles": "1.0.0-beta5",
    "Microsoft.AspNet.Tooling.Razor": "1.0.0-beta5",
    "Microsoft.Framework.Configuration.Abstractions": "1.0.0-beta5",
    "Microsoft.Framework.Configuration.Json": "1.0.0-beta5",
    "Microsoft.Framework.Configuration.UserSecrets": "1.0.0-beta5",
    "Microsoft.Framework.Logging": "1.0.0-beta5",
    "Microsoft.Framework.Logging.Console": "1.0.0-beta5",
    "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.0.0-beta5"
  },

**Comment by davidfowl** [comment]
I'd recommend reading the announcements repository while doing updates during these betas. It'll significantly cut down the time you spend here :smile:. Go here https://github.com/aspnet/Announcements/issues and filter by milestone (if you're going from beta5 to beta7 you might also want to see what happened in beta6).

You also need to update your global.json to say beta7. 

There's also new visual studio tooling associated with this release: http://www.microsoft.com/en-us/download/details.aspx?id=48738&fa43d42b-25b5-4a42-fe9b-1634f450f5ee=True

**Comment by aspnet-hello** [comment]
This issue is being closed because it has not been updated in 3 months.

We apologize if this causes any inconvenience. We ask that if you are still encountering this issue, please log a new issue with updated information and we will investigate.
